### PR TITLE
bt-160: create chat page layout

### LINKED ISSUE #160: Create chat page layout
**User story:**

As a talent, I want to have the chats page, so that I can communicate with potential employers.

**Acceptance criteria:**

- There should be the “Chats” name of the page in the upper left corner
- Chat screen should contain two parts: chat sidebar on the left side of the screen and main chat block
- The chat sidebar must include an input field for the search field on the top
- When the user chooses the search field it should be highlighted
- List of user’s chats should be displayed below the search field
- Every chat on the chat sidebar should be displayed with the user’s profile photo, user name, the last message and the last message time of delivery
- When the talent selects the chat it should be highlighted
- There should be a placeholder for the search field and the text area field
- There should be a chat page with messages in the middle with an input text area field at the bottom
- The text area field should be with the send message button
- Allowed length for text area – 1-5000 characters
- The user’s profile photo and user name should be displayed on the top of the chat page with messages

image

### PR DESCRIPTION
**Pull request by JemiJe**
- all chat components were changed [124](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/197), [126](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/199), [125](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/201), [129](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/213)
- send string is trimmed
- max message length = 1000 chars
- mock data is necessary, without it would be errors, because come props are required
- `enter / click` to send message
- `shift+enter` - new line (message input stretches up to 3 lines)
- adaptive breakpoints 1200px and 900px screen width
- choosing chat, send message and mock data were made for demo/testing
- minor fixes and improvements - **in progress**
- as I understand API and redux in progress

to test change it in `index.tsx`
```jsx
                                        path: AppRoute.CHATS,
                                        element: (
                                            // <ProtectedRoute>
                                            <PageLayout avatarUrl="" isOnline>
                                                <ChatsPage />
                                            </PageLayout>
                                            // </ProtectedRoute>
                                        ),
```

Screenshot (726)
Screenshot (727)
Screenshot (729)

### CODE REVIEW

**Comment by nikita-remeslov** [comment]
Conflicts

**Review by nikita-remeslov — APPROVED** [review]

File: frontend/src/bundles/chat/components/message-list/message-list.tsx [hunk 1]
```diff
@@ -1,4 +1,8 @@
+import { useRef } from 'react';
```
Comment (ptarho) [inline frontend/src/bundles/chat/components/message-list/message-list.tsx]: Import from hooks.js?

File: frontend/src/bundles/chat/components/chat-header/chat-header.tsx [hunk 2]
```diff
@@ -5,23 +5,36 @@ import styles from './styles.module.scss';
...
 
     return (
-        <Grid className={styles.wrapper}>
+        <Grid className={getValidClassNames(styles.wrapper, className)}>
             <Grid className={styles.logo}>
                 <Avatar isSmall={true} src={avatarUrl} alt={title} />
             </Grid>
             <Grid className={styles.info}>
-                <p className={styles.title}>{title}</p>
+                <p
```
Comment (ptarho) [inline frontend/src/bundles/chat/components/chat-header/chat-header.tsx]: Should not we use Typography?

**Comment by ptarho** [comment]
Also on 320px chat info button is too big, name of chat is overlapping with icon (screen 2)
image
image

File: frontend/src/bundles/chat/components/message-input/styles.module.scss [hunk 3]
```diff
@@ -14,7 +13,6 @@
     align-items: center;
     width: 100%;
     padding: 10px 20px;
-    padding-right: 40px;
```
Comment (vladjsp) [inline frontend/src/bundles/chat/components/message-input/styles.module.scss:17]: without this padding letters at the end ot the line will overlap send icon

**Comment by JemiJe** [comment]
> Also on 320px chat info button is too big, name of chat is overlapping with icon (screen 2)
> 
> image
> 
> image
> 
> 
> 
> 

what min screen width we should adapt? maybe 400px is enough. maybe I missed some requirements specified

File: frontend/src/bundles/chat/components/message-list/message-list.tsx (hunk 1 above)
Comment (nikita-remeslov) [reply frontend/src/bundles/chat/components/message-list/message-list.tsx]: import from common bundle

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by ptarho — APPROVED** [review]
Looks good to me

**Review by nikita-remeslov — APPROVED** [review]

**Review by vladjsp — APPROVED** [review]

**Pull request merged**